test/Spec.hs

```diff
@@ -0,0 +1,87 @@
+import Control.Exception
+import Data.List
+import System.Directory
+import System.Environment
+import System.Exit
+import System.FilePath
+import System.Posix.Temp
+import System.Which
+
+shouldEqual :: Eq a => Show a => a -> a -> IO ()
+shouldEqual a b | a /= b = do
+  putStrLn $ show a <> " does not equal " <> show b
+  exitFailure
+shouldEqual a b = putStrLn $ show a <> " == " <> show b
+
+makeExecutable :: FilePath -> IO ()
+makeExecutable f = setPermissions f . setOwnerExecutable True =<< getPermissions f
+
```

Comment:
Is there a test here that fails with the old Shelly implementation but passes now?